src/backend/columnar/columnar_customscan.c

```diff
@@ -491,15 +515,324 @@ RelationIdGetNumberOfAttributes(Oid relationId)
 }
 
 
-static Path *
-CreateColumnarScanPath(PlannerInfo *root, RelOptInfo *rel, RangeTblEntry *rte)
+/*
+ * CheckPushdownClause tests to see if clause is a candidate for pushing down
+ * into the given rel (including join clauses). This test may not be exact in
+ * all cases; it's used to reduce the search space for parameterization.
+ *
+ * Note that we don't try to handle cases like "Var + ExtParam = 3". That
+ * would require going through eval_const_expression after parameter binding,
+ * and that doesn't seem worth the effort. Here we just look for "Var op Expr"
+ * or "Expr op Var", where Var references rel and Expr references other rels
+ * (or no rels at all).
+ */
+static bool
+CheckPushdownClause(RelOptInfo *rel, Expr *clause)
+{
+	if (IsA(clause, OpExpr))
+	{
+		OpExpr *opExpr = (OpExpr *) clause;
+		Expr *lhs = list_nth(opExpr->args, 0);
+		Expr *rhs = list_nth(opExpr->args, 1);
+
+		/*
+		 * If it's not a btree operator for any opclass, it won't work. XXX:
+		 * could tighten this up to make sure it matches the opclass and
+		 * collation of the chunk group filtering predicates.
+		 */
+		if (get_op_btree_interpretation(opExpr->opno) == NIL)
+		{
+			return false;
+		}
+
+		Expr *exprSide;
+		if (IsA(lhs, Var) && ((Var *) lhs)->varno == rel->relid)
+		{
+			exprSide = rhs;
+		}
+		else if (IsA(rhs, Var) && ((Var *) rhs)->varno == rel->relid)
+		{
+			exprSide = lhs;
+		}
+		else
+		{
+			return false;
+		}
+
```

Comment:
should we have a simple `ereport` call to make sure that other side is certainly `Var`, just to be on the safe-side ?